controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTriggerer.java

```diff
@@ -0,0 +1,53 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.controller.deployment;
+
+import com.yahoo.vespa.hosted.controller.Controller;
+import com.yahoo.vespa.hosted.controller.api.integration.BuildService;
+import com.yahoo.vespa.hosted.controller.maintenance.JobControl;
+import com.yahoo.vespa.hosted.controller.maintenance.Maintainer;
+
+import java.time.Duration;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Triggers deployment jobs in an external BuildService.
+ *
+ * Triggering is performed by an Executor, as there is no guarantee the BuildService provides a timely response.
+ *
+ * @author jvenstad
+ */
+public class DeploymentTriggerer extends Maintainer {
```

Comment:
Not sure about this name, we already have `DeploymentTrigger` which considers entire deployments and `ReadyJobsTrigger` which queues jobs.

This however, only executes single jobs and doesn't know about deployments. How about `JobConsumer` or `JobExecutor`?

Reply:
DeploymentTrigger -> DeploymentScheduler and DeploymentTriggerer -> DeploymentJobTriggerer? It certainly isn't an Executor, as that is Jenkins. Consumer is far too vague.

Reply:
This is the final thing that does something with a queued job, so I think `Executor` fits. The fact that an implementation of `BuildService` may talk to a remote service is just a implementation detail in my opinion.

Reply:
Fine. I'd rename the existing Triggers to Schedulers, and keep this, but to avoid conflict and to avoid renaming and the ensuing confusion, I'll just change this instead.